main.go

```diff
@@ -434,6 +436,54 @@ func main() {
 		r.toChannelId = makeChanSet(convertChanStringToInt(params.ToChannel))
 	}
 
+	if len(params.FromNode) > 0 {
+		if len(r.fromChannelId) == 0 {
+			r.fromChannelId = map[uint64]struct{}{}
+		}
+
+		for _, node := range params.FromNode {
+
+			fromPeer, _ := hex.DecodeString(node)
+
+			// I guess we can do something better here, calling ListChannels again is not smart here
+			fromChannels, err := r.lnClient.ListChannels(infoCtx, &lnrpc.ListChannelsRequest{ActiveOnly: true, PublicOnly: true, Peer: fromPeer})
+
+			if err != nil {
+				log.Fatalf("Error fetching channels for pk: %s", err)
```

Comment:
Need a more descriptive error message, where it happened and what parameter it's related to